### ISSUE TITLE
package fails

### ISSUE THREAD

**Issue by mgkimsal** [ua-parser/uap-java#6]
-bash-4.1$ mvn package
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building ua-parser 1.3.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ ua-parser ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/me/xx/lib/uap-java/../uap-core
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ ua-parser ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.5:testResources (default-testResources) @ ua-parser ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/me/xx/lib/uap-java/../uap-core/test_resources
[INFO] skip non existing resourceDirectory /home/me/xx/lib/uap-java/../uap-core/tests
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ ua-parser ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:2.10:test (default-test) @ ua-parser ---
[INFO] Surefire report directory: /home/me/xx/lib/uap-java/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running ua_parser.CachingParserTest
Tests run: 17, Failures: 0, Errors: 17, Skipped: 0, Time elapsed: 0.125 sec <<< FAILURE!
Running ua_parser.DeviceTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.027 sec
Running ua_parser.ParserTest
Tests run: 9, Failures: 0, Errors: 9, Skipped: 0, Time elapsed: 0.013 sec <<< FAILURE!
Running ua_parser.UserAgentTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 sec
Running ua_parser.OSTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.026 sec

Results :

Tests in error:
  testCachedReplacementQuoting(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseOS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseAdditionalOS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseDevice(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseFirefox(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParsePGTS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseAll(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testCachedParseUserAgent(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testInvalidConfigThrows(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseUserAgent(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseOS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseAdditionalOS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseDevice(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseFirefox(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParsePGTS(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testParseAll(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testReplacementQuoting(ua_parser.CachingParserTest): java.io.IOException: Stream closed
  testInvalidConfigThrows(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseUserAgent(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseOS(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseAdditionalOS(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseDevice(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseFirefox(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParsePGTS(ua_parser.ParserTest): java.io.IOException: Stream closed
  testParseAll(ua_parser.ParserTest): java.io.IOException: Stream closed
  testReplacementQuoting(ua_parser.ParserTest): java.io.IOException: Stream closed

Tests run: 35, Failures: 0, Errors: 26, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.441s
[INFO] Finished at: Sun May 03 14:31:39 EDT 2015
[INFO] Final Memory: 10M/454M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.10:test (default-test) on project ua-parser: There are test failures.
[ERROR]
[ERROR] Please refer to /home/me/xx/lib/uap-java/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by bestage** [comment]
Getting the same issue, would be good if you can fix it!

**Comment by bestage** [comment]
Seems like the code is trying to load this file, which is not present in the project ,-)

private static final String REGEX_YAML_PATH = "/ua_parser/regexes.yaml";

**Comment by belerophon** [comment]
In the same directory where you have the `uap-java` repo, which you are trying to build, you need also to put this repo:
- https://github.com/ua-parser/uap-core

It contains resources `uap-java` is expecting to find.

**Comment by ameyamahajan** [comment]
Directory structure: 
> find ./ua-parser/ -depth -2
./ua-parser/
./ua-parser//uap-core
./ua-parser//uap-java

I ran command from ./ua-parser/uap-java directory, mvn package. I failed for couple of entries 
> mvn package 

Summary of mvn package 

*It fails for 
Failed tests:   testCachedParseDevice(ua_parser.CachingParserTest): Mozilla/5.0 (Linux; U; Android 4.0.3; en-us; Amaze_4G Build/IML74K) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30(..)

  testParseDevice(ua_parser.CachingParserTest): Mozilla/5.0 (Linux; U; Android 4.0.3; en-us; Amaze_4G Build/IML74K) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30(..)

  testParseDevice(ua_parser.ParserTest): Mozilla/5.0 (Linux; U; Android 4.0.3; en-us; Amaze_4G Build/IML74K) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30(..)*

**Comment by belerophon** [comment]
Yes, I'm having the same problems. It seems that the some of the different implementations are not up to date with the `core`, so some tests fail.

**Comment by ameyamahajan** [comment]
I created a test class RunTest.java, it does not execute well

> java RunTest
Exception in thread "main" java.lang.NoClassDefFoundError: org/yaml/snakeyaml/constructor/BaseConstructor
	at RunTest.main(RunTest.java:10)
Caused by: java.lang.ClassNotFoundException: org.yaml.snakeyaml.constructor.BaseConstructor
	at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 1 more


This class looks like 
>  import ua_parser.Parser;
import ua_parser.Client;
class RunTest{
  public static void main (String []args){
    try{
  String uaString = "Mozilla/5.0 (iPhone; CPU iPhone OS 5_1_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9B206 Safari/7534.48.3";
  Parser uaParser = new Parser();
  Client c = uaParser.parse(uaString);
  System.out.println(c.userAgent.family); // => "Mobile Safari"
  System.out.println(c.userAgent.major);  // => "5"
  System.out.println(c.userAgent.minor);  // => "1"
  System.out.println(c.os.family);        // => "iOS"
  System.out.println(c.os.major);         // => "5"
  System.out.println(c.os.minor);         // => "1"
  System.out.println(c.device.family);    // => "iPhone"
    }
  catch(Exception ex){ex.printStackTrace();}
  }
}


Disclaimer : I am new to maven tool.

**Comment by lefloh** [comment]
Tests are working with PR https://github.com/ua-parser/uap-java/pull/7

**Comment by knut-forkalsrud** [comment]
I noticed a commit by "commenthol" on Nov 5, 2014 tried to address the same, more or less.
That was reverted by "tobie" two days later, without further comment.

Now the regex file in ua-core has had the new formats for several months, and it seems this pull request is just waiting for someone to merge it. I'm not entirely sure who's the official maintainer here, the readme file only lists the initial author.
